auth/role.go

```diff
@@ -20,15 +20,33 @@ import (
 
 /** A group that users can belong to, with associated channel permissions. */
 type roleImpl struct {
-	Name_             string      `json:"name,omitempty"`
-	ExplicitChannels_ ch.TimedSet `json:"admin_channels,omitempty"`
-	Channels_         ch.TimedSet `json:"all_channels"`
-	Sequence_         uint64      `json:"sequence"`
-	PreviousChannels_ ch.TimedSet `json:"previous_channels,omitempty"`
+	Name_             string                  `json:"name,omitempty"`
+	ExplicitChannels_ ch.TimedSet             `json:"admin_channels,omitempty"`
+	Channels_         ch.TimedSet             `json:"all_channels"`
+	Sequence_         uint64                  `json:"sequence"`
+	PreviousChannels_ *PreviousChannelsOrRole `json:"previous_channels,omitempty"`
+	ChannelHistory_   ChannelOrRoleHistory    `json:"channel_history,omitempty"`
 	vbNo              *uint16
 	cas               uint64
 }
 
+type PreviousChannelsOrRole struct {
+	InvalSeq uint64      `json:"inval_seq"`
+	Entries  ch.TimedSet `json:"entries"`
+}
+
+type ChannelOrRoleHistory map[string]ChannelOrRoleHistoryEntries
```

Comment:
Same comment above about a more generic name, instead of ChannelOrRoleX.  Same goes for the comments describing the distinction between UpdatedAt and Seq/EndSeq.

Reply:
UpdatedAt: Timestamp when data was updated, used for pruning (not yet implemented)
Seq: Sequence when a grant was done to give access to a role/channel (only populated when endSeq is also there)
EndSeq: Sequence when access to a role/channel was revoked.